src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -423,32 +424,42 @@ def visit_Lambda(
             context.body.add_array(name, shape=shape, strides=strides, dtype=dtype)
 
         # Translate the function's body
-        result: ValueExpr | SymbolExpr = self.visit(node.expr)[0]
-        # Forwarding result through a tasklet needed because empty SDFG states don't properly forward connectors
-        if isinstance(result, ValueExpr):
-            result_name = unique_var_name()
-            self.context.body.add_scalar(result_name, result.dtype, transient=True)
-            result_access = self.context.state.add_access(result_name)
-            self.context.state.add_edge(
-                result.value,
-                None,
-                result_access,
-                None,
-                # in case of reduction lambda, the output edge from lambda tasklet performs write-conflict resolution
-                dace.Memlet(f"{result_access.data}[0]", wcr=context.reduce_wcr),
-            )
-            result = ValueExpr(value=result_access, dtype=result.dtype)
-        else:
-            result = self.add_expr_tasklet([], result.value, result.dtype, "forward")[0]
-        self.context.body.arrays[result.value.data].transient = False
-        self.context = prev_context
+        results: list[ValueExpr] = []
+        """
+        Ideally, we would like the lambdas to be single-output expressions, which implies only one child node.
+        This would simplify SDFG-generation. For this, we need a transformation pass in ITIR preprocessor
+        to unpack tuple expressions into multiple single-output expressions.
+        The same applies to scan and conditional expressions with tuples. However, for lambda expressions
+        it is easy to unpack tuples in the for-loop below, after flattening the nested list of expressions.
+        Consider removing this loop once the ITIR transformation loop is in place.
```

Comment:
I think the conclusion yesterday was that multiple-input-multiple-output lambdas are okay, we just don't want nested tuples as output. Can you perhaps update the comment to avoid the term "single-output"?